gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -210,22 +213,65 @@ func (vm *VMKeeper) newGnoTransactionStore(ctx sdk.Context) gno.TransactionStore
 }
 
 func (vm *VMKeeper) MakeGnoTransactionStore(ctx sdk.Context) sdk.Context {
-	return ctx.WithValue(gnoStoreContextKey, vm.newGnoTransactionStore(ctx))
+	return ctx.WithValue(keeperKeyGnoStore, vm.newGnoTransactionStore(ctx))
 }
 
 func (vm *VMKeeper) CommitGnoTransactionStore(ctx sdk.Context) {
 	vm.getGnoTransactionStore(ctx).Write()
 }
 
 func (vm *VMKeeper) getGnoTransactionStore(ctx sdk.Context) gno.TransactionStore {
-	txStore := ctx.Value(gnoStoreContextKey).(gno.TransactionStore)
+	txStore := ctx.Value(keeperKeyGnoStore).(gno.TransactionStore)
 	txStore.ClearObjectCache()
 	return txStore
 }
 
 // Namespace can be either a user or crypto address.
 var reNamespace = regexp.MustCompile(`^gno.land/(?:r|p)/([\.~_a-zA-Z0-9]+)`)
 
+// ExecContextCustom is a subset of [stdlibs.ExecContext] which can be used to
+// inject custom fields at genesis.
+type ExecContextCustom struct {
+	// Height is the BlockHeight: it is not recommended to set the same
+	// BlockHeight as the one used by the previous iteration of the chain, as
+	// from the realm's point of view it will eventually be as if a height
+	// occurred twice at different times: instead, use negative heights.
+	// (Last block should be -1, as 0 is the genesis block).
```

Comment:
This isn't very clear to me.  Is this saying that transactions will be executed by order of height? So, for example, -5, -4, -3, -2, -1, 0 -- genesis for unspecified transactions?

Reply:
Not necessarily. What I'm trying to communicate is that the keeper/gno.land does not actually "control" the Height; but to avoid collisions with the blocks that are created after genesis, the "space" that should be used for heights should be for negative ones.

But really, this is something that the users of this feature would need to set up with the corresponding values in genesis.json. ie. to guarantee ascending block heights, just order the transactions in a way so that they're ascending.